### PR TITLE
Update RadMon plotting

### PR DESCRIPTION
**Pull request by EdwardSafford-NOAA**
This PR includes the following changes:

- Update the GSI-monitor utility scripts to add support for the RadMon data files.
- Improve angle plotting.  The RadMon data files are now stored in tar files by data type (angle, bcoef, bcor, time) in order to avoid a file count problem on hera.  That introduced a plot problem with angle files -- the radmon_angle.tar files are quite large and file retrieval is slow (particularly on wcoss2).  This change breaks the angle plots into smaller jobs and adds more wall time if the requested number of cycles in the plot are greater than the default (121).
- Added a `--tank `option to the RadMon_IG_glb.sh script to allow override for the source of RadMon data files.
- Remove excessive log/error output.
- Removed plot support for regional data.  The RadMon has not been used on regional sources in quite a while (years).  Further, since the RadMon's data extraction files have moved to the global-workflow repository, it's not possible to directly plug in any of the monitors to regional models at the moment.  Note however that the plan for the next version of DA monitoring (monitoring 2.0) does include support for regional models.

Testing was done on wcoss2, orion, and hera using extracted RadMon data arranged in operational, rocoto workflow, and internal monitor directory structures.  Stress testing was also done by plotting with very large numbers of cycles.  Note that problems still remain with some of the very large instruments (iasi, cris) if the requested number of cycles is > ~375 (or roughly 3x the default span).  If this is found to be a problem for users then a different scheme will have to be adopted.

### CODE REVIEW

**Comment by EdwardSafford-NOAA** [comment]
Converted this to PR draft while I figure out why hdf5 now fails to build in the intel ci test.

**Comment by EdwardSafford-NOAA** [comment]
@aerorahul I hate to bug you but I need some guidance on the intel ci build failure here.  The failures begin with building the hdf5 library (line 770 in the log).  I can't see anything in the ci/spack.yaml and the .github/workflow/intel.yml files that might be causing this.  Thoughts?

I can add a bit more detail.  The intel ci test is trying to load hdf5-1.12.2.  It doesn't find a binary for that and then attempts to build hdf5, which fails.  My guess is that this is a version problem, since hpc-stack includes hdf5-1.10.6.  I've compared the GSI-monitor/ci/spack.yaml and workflow/intel.yaml to the same files in GSI and don't see a difference.  Neither one explicitly loads hdf5, but the GSI intel test is working.  

Additional update:  I created #49, a draft PR with a single, insignificant change, to run the ci tests.  Curiously both the intel and gcc tests failed.  The intel test failed in the same place, indicating that the cause of failure is not any of the changes here in #47.  That's what I expected -- this PR contains only changes to scripts, not code.  The gcc failure is confusing.  I re-ran the gcc test here in #47 and it works.  I don't know what to make of that.

**Comment by EdwardSafford-NOAA** [comment]
@DavidHuber-NOAA please take a look when you have a chance.  I'm still not sure what's up with the ci intel test failure, the actual intel builds on wcoss2, orion, and hera are fine.  Thanks.

File: src/Radiance_Monitor/image_gen/ush/RadMon_IG_glb.sh [hunk 1]
```diff
@@ -48,6 +52,7 @@ fi
 run=gdas
 pdate=""
```
Comment (DavidHuber-NOAA) [inline src/Radiance_Monitor/image_gen/ush/RadMon_IG_glb.sh:53]: The check for `$nargs` should range from 1 to 9.

**Comment by aerorahul** [comment]
@edwardhartnett @AlexanderRichert-NOAA 
This repository uses spack to build the dependencies.
It has been failing for a while.
Any assistance from the spack team is appreciated.

**Comment by AlexanderRichert-NOAA** [comment]
> @edwardhartnett @AlexanderRichert-NOAA This repository uses spack to build the dependencies. It has been failing for a while. Any assistance from the spack team is appreciated.

I'll take a look. Any idea when it started failing, or if there was a change that precipitated it?

**Comment by EdwardSafford-NOAA** [comment]
@AlexanderRichert-NOAA it first failed for me on 11/22.  Intel test only.  Today the gcc test failed when I ran a control test containing no changes to develop, though that same test ran to completion in this PR today.  The last time I can for sure say both tests ran to completion was 11/3.

**Comment by AlexanderRichert-NOAA** [comment]
@EdwardSafford-NOAA Is this run a fair example of when it was working? https://github.com/NOAA-EMC/GSI-Monitor/actions/runs/3386726189/jobs/5626502735

It's a bit of a shot in the dark, but the OS version associated with ubuntu-latest has changed recently (I think it's been going back and forth some?). The failed runs all appear to be using Ubuntu 22, and the handful of successful ones I've looked at used Ubuntu 20. Just to narrow things down, maybe you could try specifying ubuntu-20.04. If that works, it's not a good long-term solution, but we could look at intel oneapi version numbers, any intel-specific logic in hdf5/spack, etc.

**Comment by EdwardSafford-NOAA** [comment]
@AlexanderRichert-NOAA that worked.  I changed ubuntu-latest to ubuntu-20.04 and it ran to completion.  Huzah!  Thanks, and please let me know if/when I should go back to using ubuntu-latest.

**Comment by AlexanderRichert-NOAA** [comment]
@EdwardSafford-NOAA The short answer is that we want to switch back ASAP, so we should keep hunting for the root issue. I'll look at this some more today/tomorrow and see if I can figure out what might be going on.

File: src/Radiance_Monitor/image_gen/ush/mk_angle_plots.sh [hunk 2]
```diff
@@ -197,45 +197,83 @@ cd $PLOT_WORK_DIR
...
-elif [[ $MY_MACHINE = "wcoss2" ]]; then
-   $SUB -q $JOB_QUEUE -A $ACCOUNT -o ${logfile} -e ${R_LOGDIR}/plot_angle_${suffix}.err \
-	-V -l walltime=30:00 -N ${jobname} ${cmdfile}
-fi
+      echo "TASKS= $tasks"
+      if [[ ${MY_MACHINE} = "hera" || ${MY_MACHINE} = "s4" || ${MY_MACHINE} = "orion" ]]; then
+         $SUB --account ${ACCOUNT} -n ${itemctr}  -o ${logfile} -D . -J ${jobname} --time=30:00 \
+              --wrap "srun -l --multi-prog ${cmdfile}"
+
+      elif [[ ${MY_MACHINE} = "jet" ]]; then
```
Comment (DavidHuber-NOAA) [inline src/Radiance_Monitor/image_gen/ush/mk_angle_plots.sh:256]: I don't think Jet will let you submit `-n >1` to the service partition.  Did you mean to specify `-p ${BATCH_PARTITION}`?

**Review by DavidHuber-NOAA — CHANGES_REQUESTED** [review]
Here's a partial review.  I'm running a few cycles now so I can test this more thoroughly.

File: src/Radiance_Monitor/image_gen/ush/mk_angle_plots.sh (hunk 2 above)
Comment (EdwardSafford-NOAA) [reply src/Radiance_Monitor/image_gen/ush/mk_angle_plots.sh:256]: I don't have access to Jet so I will defer to you on this one.  The current version in develop does use `-p ${SERVICE_PARTITION}` with -n > 1 (typical values are 8-15), but if `-p ${BATCH_PARTITION} ` is preferable I'll make that change.

File: src/Radiance_Monitor/image_gen/ush/RadMon_IG_glb.sh (hunk 1 above)
Comment (EdwardSafford-NOAA) [reply src/Radiance_Monitor/image_gen/ush/RadMon_IG_glb.sh:53]: Done -- check for `$nargs` ranges from 1 to 9.

File: src/Radiance_Monitor/image_gen/ush/mk_angle_plots.sh (hunk 2 above)
Comment (DavidHuber-NOAA) [reply src/Radiance_Monitor/image_gen/ush/mk_angle_plots.sh:256]: You are correct, sir.  I've created a new ticket to look into this (#51) and I'll change it if needed.

**Review by DavidHuber-NOAA — CHANGES_REQUESTED** [review]
I'm having trouble running the monitor on Hera for workflow-generated stats.  It seems that `find_last_cycle.sh` is not correctly finding the last cycle.  Data is available from 2020080100 through 2020080600 and I am issuing the command
`./RadMon_IG_glb.sh test_monitor -p 2020080500 -r gdas -n 20 -t /scratch1/NESDIS/nesdis-rdo2/David.Huber/monitor`.
I receive the error
`Unable to plot, pdate is > latest_data, 2020080500,`

**Comment by EdwardSafford-NOAA** [comment]
Hmmm.  I think there's something flaky with the argument processing.  If I copy and paste to run your command I get the same error.  But if I reorder it slightly to
`./RadMon_IG_glb.sh test_radmon -p 2020080500 -r gdas -t /scratch1/NESDIS/nesdis-rdo2/David.Huber/monitor -n 20`,  I get this:
```
OK to plot
span is start_date to pdate = 2020073100, 2020080500
```
I'll take a deeper look.

**Comment by EdwardSafford-NOAA** [comment]
Oh I see the problem.  I think you have the wrong suffix in your command -- I think you want `test_radmon` instead of `test_monitor`.

**Comment by DavidHuber-NOAA** [comment]
Roger that ID-10-T error.  Thanks.

I am getting a new error now:
`Unable to set ieee_src, aborting plot`

**Comment by DavidHuber-NOAA** [comment]
This may be another error on my front.  I was only able to run 2 cycles and tried spoofing the rest of the data via copies and renames.  But without modifying contents of the ctl files, I could be throwing the plotting software off.

**Comment by EdwardSafford-NOAA** [comment]
That could be the case.  Which cycles have the real data?  And if you don't mind, can you open the file permissions so I can try running with your data?  I'd really like to rule out a bug.

**Comment by DavidHuber-NOAA** [comment]
The real data is in cycles 2020080100-2020080106.  You should have read permissions for all of the data in /scratch1/NESDIS/nesids-rdo2/David.Huber/monitor.  Do you need more than that?

I did try running on just those two cycles and I'm now receiving an error from mk_bcoef_plots.sh:

`All bcoef control files are missing from /scratch1/NESDIS/nesdis-rdo2/David.Huber/monitor/stats/test_radmon for requested date range.`

**Comment by DavidHuber-NOAA** [comment]
Ah, I see that the issue is on [line 28](https://github.com/NOAA-EMC/GSI-Monitor/blob/105cc5d501985dc3b73d3193c59b06d69ae5f2f5/src/Radiance_Monitor/image_gen/ush/mk_bcoef_plots.sh#L28).  Integer 1/4 is 0, so `ndays=0`.

**Comment by EdwardSafford-NOAA** [comment]
@DavidHuber-NOAA I implemented a fix for the ndays=0 case.  Thanks for checking a minimal data situation -- I was totally focused on the maximum case for this release.   Please take another look when you have a moment.

**Review by DavidHuber-NOAA — APPROVED** [review]
@EdwardSafford-NOAA Thanks for the fix.  I think we are good to go.  The only errors I'm seeing now are from [mk_digital_time.sh:141](https://github.com/NOAA-EMC/GSI-Monitor/blob/97f1b19117cdb0944bcdc9ff474176c1757a351f/src/Radiance_Monitor/image_gen/ush/mk_digital_time.sh#L141).  I'm getting a copious number of `cannot create regular file '.....': File exists` messages.  This doesn't really make sense to me as you have the `-f` flag, so it should be able to overwrite existing files.  I think this occurs because the script is called multiple times, so there are several attempts to move the files at the same time.  All of the files are moved in the end, so I think it's safe to continue.

**Pull request merged**